### ISSUE TITLE
sbt (0.13) junit xml reporter: "It is not a test"

### ISSUE THREAD

**Issue by TimothyKlim** [etorreborre/specs2#283]
Hi,

I have this simple spec:
```scala
import org.specs2.mutable.Specification

class MainSpec extends Specification {
  "Main".title
  "wow test" should {
    "pass" in { 1 should_== 1 }
  }
}
```
but result of junit reporter is:
```xml
<testcase classname="MainSpec" name="(It is not a test)" time="-0.001"></testcase>
```

Is this is a bug that name property is equal to "(It is not a test)"?

```scala
scalaVersion := "2.10.4"
sbtVersion := "0.3.15"
```

**Comment by francisdb** [comment]
Can this please be reopened? Seems that specs2 is outputting non-TestSelector events:

https://github.com/etorreborre/specs2/blob/master/core/src/main/scala/org/specs2/reporter/SbtPrinter.scala#L91
```
val selector = taskDef.selectors.headOption.getOrElse(new SuiteSelector)
```

but the selectors is always empty (last argument)
https://github.com/etorreborre/specs2/blob/master/core/src/main/scala/org/specs2/runner/SbtRunner.scala#L180
```
new TaskDef(arguments(0), Fingerprints.fp1, true, Array())
```

this is what the reporter expects
https://github.com/sbt/sbt/blob/0.13/testing/src/main/scala/sbt/JUnitXmlTestsListener.scala#L69
```
e.selector match {
    case selector: TestSelector => selector.testName.split('.').last
    case _ => "(It is not a test)"
}
```

**Comment by etorreborre** [comment]
@francisdb can you please try `3.2-20150326231645-b6fc1db` and tell me if that works better for you?

**Comment by francisdb** [comment]
Yes, that's better! Thanks for the quick followup

now I see this in the sbt junit file
```xml
<testcase classname="com.foo.MySpec" name="correctly calculate permutations" time="10.0">
```

compared to the one generated by specs2
```xml
<testcase name="The Bayes tool should::correctly calculate permutations" classname="com.foo.MySpec" time="0.002">
```

2 remaining issues
* there is a difference in testcase name, the should part seems to be missing
* the sbt report seems to report `10.0` test time for all test cases for all specs

there also are some other minor differences in the properties but don't care too much about those

**Comment by etorreborre** [comment]
I realised that when I had a look at the implementation this morning. I'll try to bring the listener up to the `JUnitXmlPrinter` when I have time (maybe this week-end).

**Comment by francisdb** [comment]
Ok, great!

**Comment by francisdb** [comment]
I can confirm this is fixed (tested with 3.3.1)
Nothing mentioned in the changelog?

**Comment by etorreborre** [comment]
I've mentioned it as a fix in [3.3](http://notes.implicit.ly/post/115084671904/specs2-3-3).

**Comment by skolberg** [comment]
I know, this may be to much to ask, but is there a chance to get this fixed for 2.x, too?
